### PR TITLE
feat: refactor publish.yaml to use charmhub-upload-action charm

### PR DESCRIPTION
**Pull request by ca-scribner**

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -4,31 +4,28 @@ on:
...
-            --publish-charms \
-            --publish-namespace kubeflow-charmers
-
-
+    - uses: actions/checkout@v2
+    - uses: canonical/charmhub-upload-action@initial-implementation
+      with:
+        credentials: ${{ secrets.CHARMSTORE_CREDENTIAL }}
+        charm-path: charms/${{ matrix.charm }}
+        charmcraft-channel: latest/candidate
```
Comment (DomFleischmann) [inline .github/workflows/publish.yaml:31]: This should be left with the default value, `latest/edge`

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -4,31 +4,28 @@ on:
...
-        juju-bundle publish \
-            --url cs:~kubeflow-charmers/kubeflow-pipelines \
-            --publish-charms \
-            --publish-namespace kubeflow-charmers
-
-
+    - uses: actions/checkout@v2
+    - uses: canonical/charmhub-upload-action@initial-implementation
+      with:
+        credentials: ${{ secrets.CHARMSTORE_CREDENTIAL }}
```
Comment (DomFleischmann) [inline .github/workflows/publish.yaml:29]: This credential needs to be exported with the new way of generating [tokens](https://juju.is/docs/sdk/remote-env-auth)

**Review by DomFleischmann — CHANGES_REQUESTED** [review]
The publish action needs some changes to work correctly.

**Comment by ca-scribner** [comment]
Closed because it was just used for testing

**Pull request closed without merging**